src/sentry/notifications/notifications/strategies/role_based_recipient_strategy.py

```diff
@@ -33,6 +34,21 @@ def set_member_in_cache(self, member: OrganizationMember) -> None:
         """
         self.member_by_user_id[member.user_id] = member
 
+    def get_member_role(self, user: APIUser) -> str:
+        # cache the result
+        if user.class_name() != "User":
```

Comment:
@cathyteng17 When would we get a different class?

Reply:
it looks like it's for hybrid cloud.` get_member()` also has this same logic so i wanted to keep it consistent